akka-stream/src/main/scala/akka/stream/impl/Throttle.scala

```diff
@@ -42,20 +41,21 @@ private[stream] class Throttle[T](cost: Int,
         else completeStage()
 
       override def onPush(): Unit = {
-        val timeElapsed = now() - previousTime
-        val currentTokens = Math.min(timeElapsed * speed + lastTokens, scaledMaximumBurst)
+        val currentTime = now()
+        val currentTokens = Math.min((currentTime - previousTime) * speed + lastTokens, scaledMaximumBurst)
```

Comment:
You don't need to get the time here all the time. The procedure should be:
 - calculate cost of element
 - if there are enough tokens, just remove that amount. No token refresh
 - if there are not enough tokens:
  - add those tokens to currentTokens which arrived since the last refresh timestamp
  - update the refresh timestamp
  - check if now tokens are enough or not

I.e. the idea is to calculate the time-based refreshment tokens less frequently, in larger batches. If the burst size is 4-5x larger than the average element then this saves a lot of current time evaluations.